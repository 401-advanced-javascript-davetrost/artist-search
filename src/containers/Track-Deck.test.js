import React from 'react';
import { shallow } from 'enzyme';
import TrackDeck from './Track-Deck';

import reactRouterMock from 'react-router-dom';
jest.mock('react-router-dom', () => ({}));
reactRouterMock.useParams = () => ({ id: '1234', artist: 'Grateful Dead', album: 'American Beauty' });

import useTracksMock from './use-tracks';
jest.mock('./use-tracks', () => ({}));

describe('Track Deck component', () => {

  it('renders the loading state', () => {
    useTracksMock.useTracks = () => ({ 
      loading: true,
      tracks: []
    });
    const wrapper = shallow(<TrackDeck />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a list of tracks', () => {
    useTracksMock.useTracks = () => ({ 
      loading: false, 
      tracks: [
        { id: '1', title: 'Box of Rain' },
        { id: '2', title: 'Friend of the Devil' },
        { id: '3', title: 'Attics of My Life' },
      ]
    });
    const wrapper = shallow(<TrackDeck />);
    expect(wrapper).toMatchSnapshot();
  });
});
