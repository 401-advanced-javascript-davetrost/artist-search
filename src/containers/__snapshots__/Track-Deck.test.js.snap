// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track Deck component renders a list of tracks 1`] = `
<Fragment>
  <section
    className="TrackDeck"
  >
    <h2>
      American Beauty
    </h2>
    <h3>
      by: 
      Grateful Dead
    </h3>
    <ul>
      <li
        key="1"
      >
        <TrackCard
          album="American Beauty"
          artist="Grateful Dead"
          track="Box of Rain"
        />
      </li>
      <li
        key="2"
      >
        <TrackCard
          album="American Beauty"
          artist="Grateful Dead"
          track="Friend of the Devil"
        />
      </li>
      <li
        key="3"
      >
        <TrackCard
          album="American Beauty"
          artist="Grateful Dead"
          track="Attics of My Life"
        />
      </li>
    </ul>
  </section>
</Fragment>
`;

exports[`Track Deck component renders the loading state 1`] = `
<Fragment>
  <Loading />
</Fragment>
`;
